core/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java

```diff
@@ -250,4 +260,76 @@ public void testCreateShrinkIndexFails() throws Exception {
         ensureGreen();
         assertHitCount(client().prepareSearch("target").setSize(100).setQuery(new TermsQueryBuilder("foo", "bar")).get(), 20);
     }
+
+    public void testCreateShrinkWithIndexSort() throws Exception {
+        SortField expectedSortField = new SortedSetSortField("_type", true, SortedSetSelector.Type.MAX);
+        expectedSortField.setMissingValue(SortedSetSortField.STRING_FIRST);
+        Sort expectedIndexSort = new Sort(expectedSortField);
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        prepareCreate("source")
+            .setSettings(
+                Settings.builder()
+                    .put(indexSettings())
+                    .put("sort.field", "_type")
+                    .put("sort.order", "desc")
+                    .put("number_of_shards", 8)
+                    .put("number_of_replicas", 0)
+            )
+            .addMapping("t1")
+            .get();
+        for (int i = 0; i < 20; i++) {
+            client().prepareIndex("source", "t1", Integer.toString(i))
+                .setSource("{\"foo\" : \"bar\", \"i\" : " + i + "}", XContentType.JSON).get();
+        }
+        ImmutableOpenMap<String, DiscoveryNode> dataNodes = client().admin().cluster().prepareState().get().getState().nodes()
```

Comment:
These feel fairly copy and pasted from the above. I wonder if we could make a method that both tests could call? I like that better than copy-and-paste because then I can tell what is actually different between the two test cases.